pkg/controllers/dynakube/oneagent.go

```diff
@@ -6,26 +6,16 @@ import (
 	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta1/dynakube"
 	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/oneagent"
 	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/version"
-	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/object"
-	"k8s.io/api/apps/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 func (controller *Controller) reconcileOneAgent(ctx context.Context, dynakube *dynakube.DynaKube, versionReconciler version.Reconciler) error {
-	if !dynakube.NeedsOneAgent() {
-		return controller.removeOneAgentDaemonSet(ctx, dynakube)
+	if dynakube.NeedsOneAgent() {
+		err := versionReconciler.ReconcileOneAgent(ctx, dynakube)
```

Comment:
why not just return `versionReconciler.ReconcileOneAgent(ctx, dynakube)` you don nothing with err :)